src/prpy/rave.py

```diff
@@ -235,3 +235,229 @@ def disable_padding(body, enable=False):
+   stdscr.addstr("(vi-style commands)\n")
+   stdscr.refresh()
+   while 1:
+      geom_data = body.GetLinks()[0].GetGeometries()[0].GetInfo()._vGeomData
+      r = geom_data[0]
+      h = geom_data[1]
+      c = stdscr.getch()
+      if c == ord('k'):
+         new_h = h + RESIZE_SCALE
+         resize_cylinder(env, body, height=new_h)
+         print_size(new_h, r)
+      if c == ord('j'):
+         new_h = h - RESIZE_SCALE
+         if new_h > 0.0:
+            resize_cylinder(env, body, height=new_h)
+            print_size(new_h, r)
+      if c == ord('h'):
+         new_r = r + RESIZE_SCALE
+         resize_cylinder(env, body, radius=new_r)
+         print_size(h, new_r)
+      if c == ord('l'):
+         new_r = r - RESIZE_SCALE
+         if new_r > 0.0:
+            resize_cylinder(env, body, radius=new_r)
+            print_size(h, new_r)
+      elif c == ord('q'):
+         break
+
+
+def add_sphere(env, name, radius=0.03):
+    """Add simple sphere kinbody to OpenRAVE environment"""
+    with env:
+       geom = openravepy.KinBody.Link.GeometryInfo()
+       geom._type = openravepy.GeometryType.Sphere
+       geom._vGeomData = [radius]
+       geom._vDiffuseColor = [1,0,0]
+       link = openravepy.KinBody.LinkInfo()
+       link._vgeometryinfos = [geom]
+       link._name = 'link0'
+       sphere = openravepy.RaveCreateKinBody(env, '')
+       success = sphere.Init([link], [])
+       sphere.SetName(name)
+       env.Add(sphere)
+       return sphere
+
+
+def resize_sphere(env, body, radius=None):
+    """Resize simple (single-link) sphere kinbody"""
+    with env:
+       env.Remove(body)
+       geom = body.GetLinks()[0].GetGeometries()[0].GetInfo()
+       if radius or radius == 0.0:
+          geom._vGeomData[0] = radius
+          link = body.GetLinks()[0].GetInfo()
+          link._vgeometryinfos = [geom]
+          success = body.Init([link],[])
+          env.Add(body)
+
+
+def _interactive_resize_sphere(stdscr, env, body):
+   RESIZE_SCALE = 0.01
+   stdscr.clear()
+   stdscr.addstr("'k': increase radius, 'j': decrease radius, 'q': quit\n")
+   stdscr.addstr("(vi-style commands)\n")
+   stdscr.refresh()
+   while 1:
+      r = body.GetLinks()[0].GetGeometries()[0].GetInfo()._vGeomData[0]
+      c = stdscr.getch()
+      if c == ord('k'):
+         new_r = r + RESIZE_SCALE
+         resize_sphere(env, body, radius=new_r)
+         stdscr.addstr('\rradius: {}'.format(new_r))
+      if c == ord('j'):
+         new_r = r - RESIZE_SCALE
+         if new_r > 0.0:
+            resize_sphere(env, body, radius=new_r)
+            stdscr.addstr('\rradius: {}'.format(new_r))
+      elif c == ord('q'):
+         break
+
+
+def resize_primitive_interactive(body):
+   """Live, interactive primitive object kinbody resizing
+
+   NOTE: cannot be used in a script because it creates a curses UI
+   that requires user input
+   """
+   assert type(body) is openravepy.KinBody, "body must be an OpenRAVE KinBody"
+
+   import curses.wrapper
```

Comment:
Is `curses` a standard Python module? If not, we should add it as a `exec_depend` in the `package.xml` file.